security-csrf/src/main/java/io/micronaut/security/csrf/filter/CsrfFilter.java

```diff
@@ -0,0 +1,223 @@
+               List<CsrfTokenResolver<HttpRequest<?>>> csrfTokenResolvers,
+               CsrfTokenValidator<HttpRequest<?>> csrfTokenValidator,
+               ExceptionHandler<AuthorizationException, MutableHttpResponse<?>> exceptionHandler) {
+        this.csrfTokenResolvers = csrfTokenResolvers;
+        this.futureCsrfTokenResolvers = futureCsrfTokenResolvers.isEmpty()
+                ? futureCsrfTokenResolvers
+                : FutureCsrfTokenResolver.of(csrfTokenResolvers, futureCsrfTokenResolvers);
+        this.csrfTokenValidator = csrfTokenValidator;
+        this.exceptionHandler = exceptionHandler;
+        this.csrfFilterConfiguration = csrfFilterConfiguration;
+    }
+
+    @RequestFilter
+    @Nullable
+    public Publisher<Optional<MutableHttpResponse<?>>> csrfFilter(@NonNull HttpRequest<?> request) {
+        if (!shouldTheFilterProcessTheRequestAccordingToTheUriMatch(request)) {
+            return proceedRequest();
+        }
+        if (!shouldTheFilterProcessTheRequestAccordingToTheHttpMethod(request)) {
+            return proceedRequest();
+        }
+        if (!shouldTheFilterProcessTheRequestAccordingToTheContentType(request)) {
+            return proceedRequest();
+        }
+        return futureCsrfTokenResolvers.isEmpty()
+                ? imperativeFilter(request)
+                : reactiveFilter(request);
+    }
+
+    boolean shouldTheFilterProcessTheRequestAccordingToTheUriMatch(HttpRequest<?> request) {
+        RouteMatch<?> routeMatch = request.getAttribute(HttpAttributes.ROUTE_MATCH, RouteMatch.class).orElse(null);
+        if (routeMatch instanceof UriRouteMatch<?,?> uriRouteMatch) {
+            return shouldTheFilterProcessTheRequestAccordingToTheUriMatch(uriRouteMatch);
+        }
+        return true;
+    }
+
+    boolean shouldTheFilterProcessTheRequestAccordingToTheUriMatch(UriRouteMatch<?, ?> uriRouteMatch) {
+        return shouldTheFilterProcessTheRequestAccordingToTheUriMatch(uriRouteMatch.getUri());
+    }
+
+    boolean shouldTheFilterProcessTheRequestAccordingToTheUriMatch(String uri) {
+        boolean matches = PathMatcher.REGEX.matches(csrfFilterConfiguration.getRegexPattern(), uri);
+        if (!matches) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Request uri {} does not match fitler regex pattern {}", uri, csrfFilterConfiguration.getRegexPattern());
+            }
+            return false;
+        }
+        return true;
+    }
+
+    private static Mono<Optional<MutableHttpResponse<?>>> proceedRequest() {
+        return Mono.just(Optional.empty());
+    }
+
+    private Mono<Optional<MutableHttpResponse<?>>> reactiveFilter(HttpRequest<?> request) {
+        return Flux.fromIterable(this.futureCsrfTokenResolvers)
+                .concatMap(resolver -> Mono.fromFuture(resolver.resolveToken(request))
+                        .filter(csrfToken -> {
+                            LOG.debug("CSRF Token resolved");
```

Comment:
wrap these debug statements in if statements

Reply:
resolved via https://github.com/micronaut-projects/micronaut-security/pull/1826/commits/4cf98a24962c1acd9072fe691cf0b3c2b107f009